File: output/ChatGPT/codenet/C++/Python/s526060174.py
n, m = map(int, input().split())
x = [0] * 31
for i in range(n):
    k = int(input())
    for j in range(k):
        a = int(input())
        x[a] += 1
ans = 0
for i in range(1, m+1):
    if x[i] == n:
        ans += 1
print(ans)

File: output/ChatGPT/codenet/C++/Java/s526060174.java
import java.util.Scanner;

public class s526060174 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        int m = input.nextInt();
        int[] x = new int[31];
        for (int i = 0; i < n; i++) {
            int k = input.nextInt();
            for (int j = 0; j < k; j++) {
                int a = input.nextInt();
                x[a]++;
            }
        }
        int ans = 0;
        for (int i = 1; i <= m; i++) {
            if (x[i] == n) {
                ans++;
            }
        }
        System.out.println(ans);
    }
}